service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/azure/container/CloneControlledAzureStorageContainerResourceFlight.java

```diff
@@ -0,0 +1,125 @@
+package bio.terra.workspace.service.resource.controlled.flight.clone.azure.container;
+
+import bio.terra.stairway.Flight;
+import bio.terra.stairway.FlightMap;
+import bio.terra.workspace.common.utils.FlightBeanBag;
+import bio.terra.workspace.common.utils.FlightUtils;
+import bio.terra.workspace.common.utils.RetryRules;
+import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import bio.terra.workspace.service.job.JobMapKeys;
+import bio.terra.workspace.service.resource.controlled.cloud.azure.storageContainer.ControlledAzureStorageContainerResource;
+import bio.terra.workspace.service.resource.controlled.flight.clone.CheckControlledResourceAuthStep;
+import bio.terra.workspace.service.resource.controlled.flight.clone.ClonePolicyAttributesStep;
+import bio.terra.workspace.service.resource.controlled.flight.create.GetCloudContextStep;
+import bio.terra.workspace.service.resource.controlled.flight.update.RetrieveControlledResourceMetadataStep;
+import bio.terra.workspace.service.resource.controlled.model.ControlledResource;
+import bio.terra.workspace.service.resource.model.CloningInstructions;
+import bio.terra.workspace.service.resource.model.WsmResourceType;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys;
+import bio.terra.workspace.service.workspace.model.CloudPlatform;
+import java.util.Optional;
+import java.util.UUID;
+
+public class CloneControlledAzureStorageContainerResourceFlight extends Flight {
+
+  public CloneControlledAzureStorageContainerResourceFlight(
+      FlightMap inputParameters, Object applicationContext) {
+    super(inputParameters, applicationContext);
+
+    FlightUtils.validateRequiredEntries(
+        inputParameters,
+        WorkspaceFlightMapKeys.ResourceKeys.RESOURCE,
+        JobMapKeys.AUTH_USER_INFO.getKeyName(),
+        WorkspaceFlightMapKeys.ControlledResourceKeys.CLONING_INSTRUCTIONS,
+        WorkspaceFlightMapKeys.ControlledResourceKeys.DESTINATION_RESOURCE_ID,
+        WorkspaceFlightMapKeys.ControlledResourceKeys.DESTINATION_WORKSPACE_ID);
+
+    var flightBeanBag = FlightBeanBag.getFromObject(applicationContext);
+    var sourceResource =
+        inputParameters.get(WorkspaceFlightMapKeys.ResourceKeys.RESOURCE, ControlledResource.class);
+    var userRequest =
+        inputParameters.get(JobMapKeys.AUTH_USER_INFO.getKeyName(), AuthenticatedUserRequest.class);
+    var destinationWorkspaceId =
+        inputParameters.get(
+            WorkspaceFlightMapKeys.ControlledResourceKeys.DESTINATION_WORKSPACE_ID, UUID.class);
+
+    boolean mergePolicies =
+        Optional.ofNullable(
+                inputParameters.get(WorkspaceFlightMapKeys.MERGE_POLICIES, Boolean.class))
+            .orElse(false);
+    var cloningInstructions =
+        Optional.ofNullable(
```

Comment:
At this point it shouldn't be nullable because of the similar conditional code at https://github.com/DataBiosphere/terra-workspace-manager/pull/883/files#diff-3aaa5bf7189ebec04c5d21fb75fdaed7df9f73f342118e76ecde18b2e1f69b92R281, correct?

Reply:
There should be some test coverage of the override behavior.

Reply:
See this [unit test](https://github.com/DataBiosphere/terra-workspace-manager/blob/1b725dff35e3e1667f36eb1e7aad95f19269907f/service/src/test/java/bio/terra/workspace/service/resource/controlled/flight/clone/azure/container/CloneControlledAzureStorageContainerResourceFlightTest.java#L38)